dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.PushNotifications
+{
+    // Event args for the Push payload.
+    runtimeclass PushNotificationReceivedEventArgs
+    {
+        // Initialize using the IBackgroundInstance: used specifically for the Background Activation scenario
+        static PushNotificationReceivedEventArgs CreateFromBackgroundTaskInstance(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+// Initialize using the PushNotificationEventArgs from Windows: used specifically for in-memory event handling when app is already in foreground
+static PushNotificationReceivedEventArgs CreateFromPushNotificationReceivedEventArgs(Windows.Networking.PushNotifications.PushNotificationReceivedEventArgs args);
+
+// The Push payload
+byte[] Payload { get; };
+
+// Gets a deferral to run under specific modes like low power mode
```

Comment:
Not sure how low power mode is related to deferrals.

Reply:
We want to safely retrieve the notification payload if it was received during CS.

According to the spec: 

> Care needs to be taken to Get a Deferral and Complete the Deferral while processing the payload for
> Power Management compliance.